From 7287508b986e65403b3352889c94249ee11a231a Mon Sep 17 00:00:00 2001
From: Benoit Bovy <benbovy@gmail.com>
Date: Wed, 11 Dec 2024 09:09:47 +0100
Subject: [PATCH] another approach for openssl

---
 CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5fb3e93..0028c86 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,6 +173,9 @@ elseif(${S2_SOURCE} STREQUAL "SYSTEM")
   endif()
 endif()

+# --- OpenSSL
+find_package(OpenSSL REQUIRED)
+
 # --- Abseil (bundled build not supported)

 find_package(absl REQUIRED)
@@ -271,7 +274,8 @@ if(MSVC)
   target_compile_options(s2geography PUBLIC /J)
 endif()

-target_link_libraries(s2geography PUBLIC s2::s2 absl::memory absl::str_format)
+target_link_libraries(s2geography PUBLIC s2::s2 absl::memory absl::str_format
+                                         OpenSSL::SSL OpenSSL::Crypto)

 # Set somewhat aggressive compiler warning flags
 if(S2GEOGRAPHY_EXTRA_WARNINGS)
--
2.36.0
